pkg/k8s/watchers/watcher.go

```diff
@@ -177,6 +178,11 @@ type EgressGatewayManager interface {
 	OnDeleteNode(node nodeTypes.Node)
 }
 
+type WorldCIDRsManager interface {
+	OnAddWorldCIDRSet(cidrSet worldcidrs.CIDRSet)
+	OnDeleteWorldCIDRSet(id worldcidrs.CIDRSetID)
+}
+
```

Comment:
Instead of hooking into the existing K8sWatcher, it's better to use the new `Resource[T]`-style K8s watcher framework. See the CNP watcher as an example.

Reply:
cc @cilium/sig-foundations, I think it would be valuable to have some small write up or something to link to, to aid developers when adding new K8s watchers.

Reply:
I think there's https://docs.cilium.io/en/latest/contributing/development/hive/#kubernetes-resource-and-store, but it could certainly be expanded. https://github.com/cilium/cilium/tree/main/pkg/k8s/resource/example is an example of how to use it.